presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftRangeValueSet.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.presto.spi.predicate.Marker;
+import com.facebook.presto.spi.predicate.Range;
+import com.facebook.presto.spi.predicate.SortedRangeSet;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftEnum;
+import com.facebook.swift.codec.ThriftEnumValue;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.Bound.fromMarkerBound;
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.toRange;
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static java.util.stream.Collectors.toList;
+
+@ThriftStruct
+public final class ThriftRangeValueSet
```

Comment:
Some of these `Thrift`-prefixed structures directly correspond to internal ones used by Presto engine. For this reason their names closely match their non-thrift counterparts. Same is true for other comments you made below.